panic runtime SIGSEGV

### Description

Mac M4

Using the following configuration:

With
`LIMA_SSH_PORT_FORWARDER=false`

```
portForwards:
  - guestSocket: "tcp:2222"
    hostSocket: "tcp:2222"    
```

```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x80 pc=0x10553b134]

goroutine 1 [running]:
main.shellAction(0x14000675808, {0x1400066f6e0?, 0x4?, 0x1055544c7?})
	/private/tmp/lima-20241227-4789-ev5r3o/lima-1.0.3/cmd/limactl/shell.go:94 +0x3c4
github.com/spf13/cobra.(*Command).execute(0x14000675808, {0x1400066f6b0, 0x1, 0x1})
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0x834
github.com/spf13/cobra.(*Command).ExecuteC(0x14000674c08)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
	/private/tmp/lima-20241227-4789-ev5r3o/lima-1.0.3/cmd/limactl/main.go:28 +0x20
```

Can confirm this is still happening in `lima v2.1.4` under an unhandled YAML parsing error path. 

In my case, it was a simple indentation typo under the `qemu` key inside `lima.yaml`:
```sh
bash-5.1$ limactl shell alpine
WARN[0000] Non-strict YAML detected; please check for typos  comment="main file `/home/dio/.lima/alpine/lima.yaml`" error="[11:1] unknown field \"qemu\"\n   8 | video:\n   9 |   display: \"default\"\n  10 | \n> 11 | qemu:\n       ^\n  12 |   vga: \"virtio\"\n  13 |   tablet: true\n  14 | \n  15 | "
alpine-vm:~$ exit
bash-5.1$ vi +13 ~/.lima/alpine/lima.yaml 
bash-5.1$ grep -A3 -n qemu ~/.lima/alpine/lima.yaml
11:qemu:
12-  vga: "virtio"
13-    tablet: true # <--- wrong indentation
14-
bash-5.1$ limactl stop -f alpine
INFO[0000] The  driver process seems already stopped    
INFO[0000] The host agent process seems already stopped 
INFO[0000] Removing *.pid *.sock *.tmp under "/home/dio/.lima/alpine" 
INFO[0000] Removing "/home/dio/.lima/alpine/ga.sock"    
INFO[0000] Removing "/home/dio/.lima/alpine/ha.pid"     
INFO[0000] Removing "/home/dio/.lima/alpine/ha.sock"    
INFO[0000] Removing "/home/dio/.lima/alpine/qemu.pid"   
INFO[0000] Removing "/home/dio/.lima/alpine/qmp.sock"   
INFO[0000] Removing "/home/dio/.lima/alpine/serial.sock" 
INFO[0000] Removing "/home/dio/.lima/alpine/serialv.sock" 
INFO[0000] Removing "/home/dio/.lima/alpine/ssh.sock"
```

When I tried to start the instance, `limactl` crashed with `SIGSEGV` inside `loadOrCreateInstance`:

```logs
INFO[0000] Using the existing instance `alpine`         
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x200 pc=0x120cede]

goroutine 1 [running]:
main.loadOrCreateInstance(0x340b46e7c908, {0x340b46ebc670?, 0x0?, 0x0?}, 0x0)
	/var/tmp/lima-20260703-10088-hm9vqo/lima-2.1.4/cmd/limactl/start.go:290 +0x83e
main.startAction(0x340b46e7c908, {0x340b46ebc670, 0x1, 0x1})
	/var/tmp/lima-20260703-10088-hm9vqo/lima-2.1.4/cmd/limactl/start.go:576 +0x7d
github.com/spf13/cobra.(*Command).execute(0x340b46e7c908, {0x340b46ebc640, 0x1, 0x1})
	/home/linuxbrew/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1015 +0xb14
github.com/spf13/cobra.(*Command).ExecuteC(0x340b46e7c308)
	/home/linuxbrew/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
	/home/linuxbrew/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1071
main.main()
	/var/tmp/lima-20260703-10088-hm9vqo/lima-2.1.4/cmd/limactl/main.go:51 +0xca
```

The underlying issue is definitely that `limactl` doesn't halt early enough when the configuration file fails to unmarshal properly, leading to a nil-pointer dereference further down the line. (Environment: **Slackware 15.0 w/ Linuxbrew built binary**).

--- a/cmd/limactl/start.go
+++ b/cmd/limactl/start.go
@@ -284,6 +284,16 @@ func loadOrCreateInstance(cmd *cobra.Command, args []string, createOnly bool) (*
 		// store.Inspect() will validate the template name (in case it has been set to arg)
 		inst, err := store.Inspect(ctx, tmpl.Name)
 		if err == nil {
+			configErr := errors.Join(inst.Errors...)
+			if configErr != nil {
+				logrus.WithError(configErr).Errorf("Instance %q has configuration errors", tmpl.Name)
+			}
+			if inst.Config == nil {
+				if configErr != nil {
+					return nil, fmt.Errorf("instance %q has configuration errors: %w", tmpl.Name, configErr)
+				}
+				return nil, fmt.Errorf("instance %q has no configuration", tmpl.Name)
+			}
 			if createOnly {
 				return nil, fmt.Errorf("instance %#q already exists", tmpl.Name)
 			}

--- a/cmd/limactl/start_test.go
+++ b/cmd/limactl/start_test.go
@@ -1,0 +1,70 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"gotest.tools/v3/assert"
+
+	"github.com/lima-vm/lima/v2/pkg/limatype/filenames"
+)
+
+func TestLoadOrCreateInstance(t *testing.T) {
+	tests := []struct {
+		name            string
+		yaml            string
+		wantCPUs        int
+		wantErrContains []string
+	}{
+		{
+			name:     "valid configuration",
+			yaml:     "images: [{location: /}]\ncpus: 2\n",
+			wantCPUs: 2,
+		},
+		{
+			name: "invalid configuration",
+			yaml: "arch: [invalid\n",
+			wantErrContains: []string{
+				`instance "test" has configuration errors`,
+				"failed to unmarshal YAML",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			limaDir := t.TempDir()
+			t.Setenv("LIMA_HOME", limaDir)
+			instanceDir := filepath.Join(limaDir, "test")
+			assert.NilError(t, os.MkdirAll(instanceDir, 0o700))
+			assert.NilError(t, os.WriteFile(filepath.Join(instanceDir, filenames.LimaYAML), []byte(tt.yaml), 0o600))
+
+			inst, err := loadOrCreateInstance(newTestStartCommand(t), []string{"test"}, false)
+			if len(tt.wantErrContains) > 0 {
+				for _, want := range tt.wantErrContains {
+					assert.ErrorContains(t, err, want)
+				}
+				assert.Assert(t, inst == nil)
+				return
+			}
+
+			assert.NilError(t, err)
+			assert.Assert(t, inst != nil)
+			assert.Assert(t, inst.Config != nil)
+			assert.Equal(t, *inst.Config.CPUs, tt.wantCPUs)
+		})
+	}
+}
+
+func newTestStartCommand(t *testing.T) *cobra.Command {
+	t.Helper()
+	cmd := newStartCommand()
+	cmd.Flags().Bool("tty", false, "")
+	cmd.SetContext(t.Context())
+	return cmd
+}

